packages/firestore/src/remote/persistent_stream.ts

```diff
@@ -253,6 +245,108 @@ export abstract class PersistentStream<
   }
 
   /**
+   * Initializes the idle timer. If no write takes place within one minute, the
+   * WebChannel stream will be closed.
+   */
+  markIdle(): void {
+    this.idle = true;
+    this.queue
+      .schedule(() => {
+        return this.handleIdleCloseTimer();
+      }, IDLE_TIMEOUT_MS)
+      .catch((err: FirestoreError) => {
+        // We ignore Promise rejections for cancelled idle checks.
+        assert(
+          err.code === Code.CANCELLED,
+          `Received unexpected error in idle timeout closure. Expected CANCELLED, but was: ${err}`
+        );
+      });
+  }
+
+  /** Sends a message to the underlying stream. */
+  protected sendRequest(msg: SendType): void {
+    this.cancelIdleCheck();
+    this.stream!.send(msg);
+  }
+
+  /** Called by the idle timer when the stream should close due to inactivity. */
+  private handleIdleCloseTimer(): Promise<void> {
+    if (this.isOpen() && this.idle) {
+      // When timing out an idle stream there's no reason to force the stream into backoff when
+      // it restarts so set the stream state to Initial instead of Error.
+      return this.close(PersistentStreamState.Initial);
+    }
+    return Promise.resolve();
+  }
+
+  /** Marks the stream as active again. */
+  private cancelIdleCheck() {
+    this.idle = false;
+  }
+
+  /**
+   * Closes the stream and cleans up as necessary:
+   *
+   * <ul>
```

Comment:
Nit: Typescript comments aren't thinly veiled HTML the way Javadoc comments are. Do this as a markdown list, like we did in [mutation.ts](https://github.com/firebase/firebase-js-sdk/blob/deedb015d3bd7b63dca10459f87180a73e5c77a9/packages/firestore/src/model/mutation.ts#L333).

Reply:
Adjusted. Thanks for pointing this out.